libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -620,11 +622,18 @@ internal class MapboxTripSession(
                     }
                 }
             }
-            checkVoiceInstructionEvent(progress.voiceInstructions) { voiceInstruction ->
-                voiceInstructionsObservers.forEach {
-                    it.onNewVoiceInstructions(voiceInstruction)
-                }
+        }
+    }
+
+    private fun triggerVoiceInstructionEvent(progress: RouteProgress?, status: NavigationStatus) {
+        if (progress == null) return
+        val voiceInstructions = progress.voiceInstructions
+        val navigatorTriggeredNewInstruction = status.voiceInstruction != null
+        if (voiceInstructions != null && navigatorTriggeredNewInstruction) {
+            voiceInstructionsObservers.forEach {
+                it.onNewVoiceInstructions(voiceInstructions)
             }
+            lastVoiceInstruction = progress.voiceInstructions
```

Comment:
```suggestion
        ifNonNull(routeProgress.voiceIntruction, (status.voiceInstruction != null)) { upcomingVoiceInstructions, isTriggeredNewInstruction -> 
          if(isTriggeredNewInstruction) { 
             voiceInstructionsObservers.forEach {
                it.onNewVoiceInstructions(upcomingVoiceInstructions)
            }
            lastVoiceInstruction = upcomingVoiceInstructions
          }
        }
```

Reply:
don't you mind if I leave the code as is? I'm not a big fan of syntax sugar if it doesn't reduce lines count. BTW if you're insisting let's discuss it 🙂

Reply:
I also like the current version more. For me a simple `if` check looks cleaner than these custom `ifNonNull` functions.